README.md

```diff
@@ -1,2 +1,179 @@
+# Hipo Frontend TypeScript Project Template
+
+[![dependencies][dependencies-image] ][dependencies-url]
+[![devdependencies][devdependencies-image] ][devdependencies-url]
+
+This template includes all the configurations for kick-starting a project with React and TypeScript.
+
+## Table of Contents
+
+- [Setup](#setup)
+- [Development and Releases](#development-and-releases)
+- [GIT Hooks](#git-hooks)
+- [Changelog Generation](#changelog-generation)
+- [More Information](#more-information)
+
+## Setup
+
+For now, you need to manually copy-paste all the files from the template into your directory. Eventually, there will be a command-line tool that does this for you. After you copied all the files in this template, you will need to take the following steps:
+
+- This project requires Node >= 10.x and npm >= 6.x for proper development environment.
+
+1. Update necessary fields inside `package.json` for your projects.
+
+2. Go over all the files inside `config` folder and update them with your project's necessities. Here you will find the following configuration files:
+
+- `apiConfig:` Configurations for the API requests in different environments.
+- `assetConfig:` Paths for the assets in different environments
+- `s3Config:` S3 configurations for different environments.
+
+3. Update `changelog-context-variables.json` to include your project's `codebaseProjectName`.
+
+4. Run `npm install` to install dependencies.
+
+5. Check commented out code for Sentry integration. You will find them inside `index.tsx` and `NetworkManager.ts`. Make sure to check `sentryHandler.ts` for correcting integration.
+
+## Development and Releases
+
+### Starting the development server
+
+```shell
+npm start
+```
+
+This will build out the frontend for development environment. It is an alias for `npm run build:dev:live`. This will use the `webpack.dev.config.js` file and start out the webpack dev server with `local` being the target environment. If you want to build in another environment then do `npm run build:dev:live -- --env.target=TARGET_ENV` where TARGET_ENV may be one of the following: production, staging or local (or preproduction, if you add it). You can simply run `npm run build:dev` if you don't want to listen for changes.
+
+It is recommended to split your terminal and run `npm start` in one and `npm run type-check:watch` in another. `npm run type-check` offers cleaner output when you have errors from the `tsc`.
+
+### Releases
+
+```shell
+npm run build:release -- --env.target=RELEASE_TARGET --env.awsAccessKeyId=AWS_ACCESS_KEY_ID_FOR_YOUR_TARGET_ENV --env.awsSecretAccessKey=AWS_SECRET_ACCESS_KEY_FOR_YOUR_TARGET_ENV
+```
+
+This will generate `/build` folder with static files to serve. `RELEASE_TARGET` may be one of the following: production, staging or local (or preproduction, if you add it). During server deployment (executed by the CI jobs), RELEASE_TARGET is derived by the branch name (currently can be staging, or production).
```

Comment:
Should we add sample `.circleci/config.yml`?

Reply:
Yes thats in my todo